src/community/jdbcstore/src/main/java/org/geoserver/jdbcstore/internal/JDBCQueryHelper.java

```diff
@@ -0,0 +1,568 @@
+        try (Connection c = ds.getConnection()) {
+            LOGGER.log(Level.FINEST, query.toString());
+            PreparedStatement stmt = query.toStatement(c);
+            
+            try {            
+                ResultSet rs = stmt.executeQuery();
+                
+                try {
+                    if(rs.next()) {                
+                        assert(rs.last());
+                        Map<String, Object> result = new HashMap<String, Object>();
+                        for(int i = 0; i< fields.length; i++) {
+                            result.put(fields[i].getFieldName(), fields[i].getValue(rs));
+                        }
+                        return result;
+                    } else {
+                        return null;
+                    }
+                } finally {
+                    rs.close();
+                }
+            } finally {
+                stmt.close();
+            }
+        } catch (SQLException ex) {
+            throw new IllegalStateException("SelectQuery Failed",ex);
+        }
+    }
+    
+    public int anyUpdateQuery(QueryBuilder query) {
+        try (Connection c = ds.getConnection()) {
+            LOGGER.log(Level.FINEST, query.toString());
+            PreparedStatement stmt = query.toStatement(c);
+            try {
+                return stmt.executeUpdate();
+            } finally {
+                stmt.close();
+            }
+        } catch (SQLException ex) {
+            throw new IllegalStateException("UpdateQuery Failed",ex);
+        }
+    }
+    
+    public List<Integer> anyInsertQuery(QueryBuilder query) {
+        try (Connection c = ds.getConnection()) {
+            LOGGER.log(Level.FINEST, query.toString());
+            PreparedStatement stmt = query.toStatement(c, Statement.RETURN_GENERATED_KEYS);
+            try {
+                if (stmt.executeUpdate() <= 0) {
+                    return null;
+                }
+                
+                ResultSet rs = stmt.getGeneratedKeys();
+                List<Integer> list = new ArrayList<Integer>();
+                while (rs.next()) {
+                    list.add(rs.getInt(1));
+                }
+                rs.close();
+                return list;
+            } finally {
+                stmt.close();
+            }
+        } catch (SQLException ex) {
+            throw new IllegalStateException("InsertQuery Failed", ex);
+        }
+    }
+        
+    /**
+     * 
+     * Blobs should always be queried with this method, not with the regular selects!
+     * 
+     */
+    public InputStream anyBlobQuery(QueryBuilder query, Field<InputStream> field) {
+        Connection c;        
+        boolean closeConnection = false;
+        try {
+            c = ds.getConnection();
+        } catch (SQLException ex) {
+            throw new IllegalStateException("Could not connect to DataSource.",ex);
+        } 
+        try {
```

Comment:
No, not applicable here. Note what the method actually does: it returns an Inputstream that reads a blob from the db. The connection needs to remain open as long as the Inputstream is open. It is closed automatically when the inputstream is closed.